src/helpers.ts

```diff
@@ -113,6 +122,20 @@ export function interpretIdentifier(identifier: string): { address: string; publ
   }
 }
 
+export async function signMetaTxData(
+  identity: string,
+  signerAddress: string,
+  privateKeyBytes: Uint8Array,
+  dataBytes: Uint8Array,
+  didReg: Contract
+) {
+  const nonce = await didReg.nonce(signerAddress)
```

Comment:
This method should be tested against the [existing deployments](https://github.com/uport-project/ethr-did-registry#contract-deployments) because there are differences between the older contract ([0xdca7ef03e98e0dc2b855be647c39abe984fcf21b](https://etherscan.io/address/0xdca7ef03e98e0dc2b855be647c39abe984fcf21b)) and the latest one (for example [0x63eD58B671EeD12Bc1652845ba5b2CDfBff198e0](https://explorer.mainnet.aurora.dev/address/0x63eD58B671EeD12Bc1652845ba5b2CDfBff198e0)) regarding **`nonce`** tracking.

See https://github.com/uport-project/ethr-did-registry/pull/12

This issue in particular makes me want to move to a new contract.

Reply:
So is this about staying compatible to different versions of the contract?